garnbarn_api/views.py

```diff
@@ -1,28 +1,128 @@
+            return Response({
+                "message": "User already linked the account with LINE"
+            }, status=status.HTTP_400_BAD_REQUEST)
+        check_list = ["code", "clientId", "redirectUri"]
+        for item in check_list:
+            if not request_payload["credential"].get(item):
+                return Response({
+                    "message": f"To link account with LINE, Field `{item}` in credential is required"
+                })
+        credential = request_payload["credential"]
+        line_login = LineLoginPlatformHelper()
+        try:
+            line_login.verify_login_code(
+                credential["code"], credential["redirectUri"], credential["clientId"])
+            line_profile = line_login.get_user_profile()
+        except LineApiError as e:
+            return Response(e.line_error_object, status=status.HTTP_400_BAD_REQUEST)
+        request.user.line = line_profile["userId"]
+        request.user.save()
+        return Response({}, status=status.HTTP_200_OK)
+
+    @action(methods=['POST'], detail=False,
+            url_path='unlink', url_name='unlink')
+    def unlink(self, request):
+        try:
+            request_payload = json.loads(request.body)
+        except json.JSONDecodeError:
+            return Response({
+                "message": "The body contain invalid json format."
+            }, status=status.HTTP_400_BAD_REQUEST)
+        if request_payload.get("platform") != "line":
+            return Response({
+                "message": "You didn't specify the platform or the platform you specify is not supported."
+            }, status=status.HTTP_400_BAD_REQUEST)
+        if not request.user.line:
+            return Response({
+                "message": "User is not linked the LINE account"
+            }, status=status.HTTP_400_BAD_REQUEST)
+        request.user.line = None
+        request.user.save()
+        return Response({}, status=status.HTTP_200_OK)
 
 
 class AssignmentViewset(viewsets.ModelViewSet):
     authentication_classes = [FirebaseAuthIDTokenAuthentication]
     permission_classes = [IsAuthenticated]
-    serializer_class = garnbarn_serializer.CreateAssignmentApiSerializer
+    serializer_class = AssignmentSerializer
 
     def get_queryset(self):
+        user_data = self.request.user.uid
+
         if self.request.query_params.get('fromPresent') == "true":
-            data = Assignment.objects.exclude(
+            assignment = Assignment.objects.exclude(
                 due_date__lt=date.today())
-            data = data.exclude(due_date=None).order_by('due_date')
+            assignment = assignment.exclude(due_date=None).order_by('due_date')
         else:
-            data = Assignment.objects.get_queryset().order_by('id')
-        return data
+            assignment = Assignment.objects.get_queryset().filter(
+                Q(author=user_data) | Q(tag__subscriber__icontains=user_data)).order_by('id')
+        return assignment
```

Comment:
What Q does?

Reply:
@SoSorryTT

Reply:
Since he still doesn't answer. I'll take the place to answer since this PR is very late now. I think it is just an object that encapsulates the SQL expression. Here is the detail of the [Q Object](https://micropyramid.com/blog/querying-with-django-q-objects/)